### PR TITLE
Disable HiveTextReaders for CDH due to NVIDIA#7423

### PR DESCRIPTION
**Pull request by sririshindra**
Signed-off-by: Rishi <sririshindra@gmail.com>

CDH spark a little more robust/forgiving in reading certain inputs as compared to to upstream spark. spark-rapids follows upstream spark conventions and as such cdh users might see a discrepancy when reading an input on CPU vs GPU. In order to avoid that we are falling back on the CPU when Hive delimited text table reads are involved.

### CODE REVIEW

File: sql-plugin/src/main/330cdh/scala/org/apache/spark/sql/hive/cdh/HiveOverrides.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2019-2023, NVIDIA CORPORATION.
```
Comment (mythrocks) [inline sql-plugin/src/main/330cdh/scala/org/apache/spark/sql/hive/cdh/HiveOverrides.scala:2]:
```suggestion
 * Copyright (c) 2023, NVIDIA CORPORATION.
```

**Comment by sameerz** [comment]
build

File: sql-plugin/src/main/330cdh/scala/org/apache/spark/sql/hive/cdh/HiveOverrides.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+package org.apache.spark.sql.hive.cdh
+
+import com.nvidia.spark.rapids.{ExecChecks, GpuExec, GpuOverrides, SparkPlanMeta, TypeSig}
+
+import org.apache.spark.sql.hive.execution.HiveTableScanExec
+
+object HiveOverrides {
+
+  def getHiveTableExecRule() = {
```
Comment (mythrocks) [inline sql-plugin/src/main/330cdh/scala/org/apache/spark/sql/hive/cdh/HiveOverrides.scala:24]: We should probably mention that this is a workaround for the mismatches listed in #7423.

**Review by mythrocks — CHANGES_REQUESTED** [review]
A couple of minor nitpicks, but this is looking good to me.

File: sql-plugin/src/main/330cdh/scala/org/apache/spark/sql/hive/cdh/HiveOverrides.scala (hunk 1 above)
Comment (sririshindra) [reply sql-plugin/src/main/330cdh/scala/org/apache/spark/sql/hive/cdh/HiveOverrides.scala]: Done

**Review by mythrocks — APPROVED** [review]

**Comment by mythrocks** [comment]
Thanks for fixing this, @sririshindra. I've merged this.

**Pull request merged**